python/mxnet/module/module.py

```diff
@@ -171,6 +174,77 @@ def save_checkpoint(self, prefix, epoch, save_optimizer_states=False):
             self.save_optimizer_states(state_name)
             logging.info('Saved optimizer state to \"%s\"', state_name)
 
+    def export_serving(self, prefix, epoch, signature, save_optimizer_states=False,
+                       use_synset=False):
+        """Export current progress to a zip file used by MXNet model serving.
+
+        Parameters
+        ----------
+        prefix : str
+            The file prefix to export to.
+        epoch : int
+            The current epoch number.
+        signature : dict
+            A dictionary containing model input and output information.
+            Data names or data shapes of inputs and outputs can be automatically
+            collected from module. User needs to specify inputs and outputs MIME
+            type for http request. Currently only 'image/jpeg' and 'application/json'
+            are supported. All inputs should have the same type. This also applies for
+            outputs.
+            An example signature would be:
+                signature = { "input_type": "image/jpeg", "output_type": "application/json" }
+        save_optimizer_states : bool
+            Whether to save optimizer states to continue training.
+        use_synset : bool
+            Whether to include synset file for classification model. The synset.txt should be
+            in the same directory of training script.
+
+        Examples
+        --------
+        >>> signature = { "input_type": "image/jpeg", "output_type": "application/json" }
+        >>> model.export_serving(prefix='resnet-18', epoch=100, signature=signature)
+        Exported model to "resnet-18.zip"
```

Comment:
Let's please not use .zip as the model extension. The fact that it's zip is an implementation detail, there's no reason to expose it this way. (just as JAR is a Java Archive which is a zip)
Can we use the .dlmodel extension? (for deep learning model)

Reply:
How about .export which hints user this is an exported file. .dlmodel is kind of general name for deep learning model.